### DISCUSSION TITLE
Better support for TailwindCSS in template

### DISCUSSION THREAD

**Discussion by zizther** [solspace/craft-freeform#263] [Feature Requests]
I have noticed in the TailwindCSS template there are some instances where the class name is being put together dynamically which stops Tailwind from being able to find the class to add to the stylesheet.

A couple of examples are:

**Submit/Save button position**
`{% set columnClass = columnClass ~ " text-" ~ field.position %}`

In this instance I have converted this to a switch

```
{% switch field.position %}
    {% case "center" %}
        {% set positionClass = 'text-center' %}
    {% case "right" %}
        {% set positionClass = 'text-right' %}
    {% default %}
        {% set positionClass = 'text-left' %}
{% endswitch %}

{% set columnClass = columnClass ~ " " ~ positionClass %}
```

**CC Details**
`{% set columnClass = columnClass ~ " md:w-1/" ~ columnCount ~ " lg:w-1/" ~ columnCount ~ " xl:w-1/" ~ columnCount %}`

### TailwindCSS 3
With TailwindCSS v3 pseudo styles can be controlled via classes, this would allow you to remove the `required` css styles. You could do this by using `after:`
`{% set labelClass = labelClass ~ (field.required ? " required after:content-['*'] after:text-red-500 after:ml-2" : "") %}`

This would only work in v3, not v2 though, for backward compatibility you could keep this as is for now and and change it in future update.

**Comment by kjmartens** [comment]
Thanks for suggesting this @zizther. We'll review this and carefully consider making this change.

Please note that the sample formatting templates are meant as starting points. They attempt to cater to a wide range of customers and their wide range of use-cases, but definitely will not be adequate for all. By creating a new custom formatting template based on one of these existing samples, you will have full control over it and also eliminate the risk of one of the core formatting templates getting an unexpected update in a future release of Freeform. 🙂

**Reply by zizther** [reply]
Agreed these are a good starting place. The updates to the `Submit/Save` button position and `CC Details` help support TailwindCSS and provide a suitable template covering the fields you have in your template, for anyone who wants to use it to build on from.

I am happy to submit a PR if that would be useful.

**Reply by kjmartens** [reply]
Sure, thank you! 🙂
